packages/aws-cdk/lib/api/hotswap/ecs-services.ts

```diff
@@ -167,13 +189,49 @@ export async function isHotswappableEcsServiceChange(
   return ret;
 }
 
+function mergeTaskDefinitions(patch: {containerDefinitions: {[key:string]: any}[]}, target: AWS.ECS.DescribeTaskDefinitionResponse) {
```

Comment:
It would be slightly nicer (imo) to have this return a new object, instead of mutating the existing one in-place.

Reply:
Now I deep-copy the `target` argument. `target = JSON.parse(JSON.stringify(target));`